lib/streamlit/elements/markdown.py

```diff
@@ -64,15 +64,14 @@ def markdown(
               ``:blue-background[your text here]``.
 
         unsafe_allow_html : bool
-            By default, any HTML tags found in the body will be escaped and
-            therefore treated as pure text. This behavior may be turned off by
-            setting this argument to True.
+            Whether to render HTML within ``body``. If this is ``False``
+            (default), any HTML tags found in ``body`` will be escaped and
+            therefore treated as raw text. If this is ``True``, any HTML
+            expressions within ``body`` will be rendered.
 
-            That said, we *strongly advise against it*. It is hard to write
-            secure HTML, so by using this argument you may be compromising your
-            users' security. For more information, see:
-
-            https://github.com/streamlit/streamlit/issues/152
+            Adding custom HTML to your app impacts safety, styling, and
+            maintainability. If you only want to insert HTML or CSS without
+            Markdown text, we recommend using ``st.html`` instead.
```

Comment:
nit: maybe we could even add the recommendation into a note to make it more visual. I think we want users to migrate away from `unsafe_allow_html`